internal/pbinfo/comments.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pbinfo
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
+const (
+	// SourcePathTopItemLength the length of a top-level type path
+	SourcePathTopItemLength = 2
+	// SourcePathTopItemTypeIndex the index of the source code
+	// item path type
+	SourcePathTopItemTypeIndex = 0
+	// SourcePathTopItemIndex the index of a source code path
+	// referncing a specific Message
+	SourcePathTopItemIndex = 1
+	// SourcePathSubItemLength the length of a source code path
+	// referencing a sub-item of a Message or Service
+	SourcePathSubItemLength = 4
+	// SourcePathSubItemTypeIndex the index of the sub-item type
+	// in a source code path
+	SourcePathSubItemTypeIndex = 2
+	// SourcePathSubItemIndex the index of a source code path
+	// referencing a specific Field
+	SourcePathSubItemIndex = 3
+	// SourcePathServiceTypeValue the value that indicates a
+	// a source code path points to a Service
+	SourcePathServiceTypeValue = 6
+	// SourcePathMessageTypeValue the value that indicates a
+	// source code path points to a Message
+	SourcePathMessageTypeValue = 4
+	// SourcePathMethodTypeValue the value the indicates a
+	// source code path points to a Method
+	SourcePathMethodTypeValue = 2
+	// SourcePathFieldTypeValue represents the value associated
+	// with the Field type in a source code path
+	SourcePathFieldTypeValue = 2
+	// SourcePathExtensionTypeValue represents the value associated
+	// with the Extension type in a source code path
+	SourcePathExtensionTypeValue = 7
+	// SourcePathEnumTypeValue represents the value associated
+	// with the Enum type in a source code path
+	SourcePathEnumTypeValue = 5
+	// SourcePathEnumValueTypeValue represents the value associated
+	// with the Enum Value type in a source code path
+	SourcePathEnumValueTypeValue = 2
+)
+
+func addComments(comments map[string]string, f *descriptor.FileDescriptorProto, loc *descriptor.SourceCodeInfo_Location) {
+	var key string
+	p := loc.Path
+
+	switch {
+	// Service comment
+	case len(p) == SourcePathTopItemLength &&
```

Comment:
Perhaps I'm running counter to modern programming practice, but I think using plain numbers would be better here.

```
case len(p) == 2 && p[0] == 6:
  f.Service[p[1]]
```
obviously won't go out of bound, and I still know that 6 refers to service because you write `f.Service` below.

Reply:
I'm cool with using plain numbers here because you provide a good explanation, but yes this implementation was to avoid magic numbers.

I would still like to leave the comment on each case, if there is no objection there.